src/registrar/views/domain.py

```diff
@@ -41,6 +42,47 @@ class DomainView(DomainPermissionView):
     template_name = "domain_detail.html"
 
 
+class DomainOrgNameAddressView(DomainPermissionView, FormMixin):
+    """Organization name and mailing address view"""
+
+    model = Domain
+    template_name = "domain_org_name_address.html"
+    context_object_name = "domain"
+    form_class = DomainOrgNameAddressForm
+
+    def get_form_kwargs(self, *args, **kwargs):
+        """Add domain_info.organization_name instance to make a bound form."""
+        form_kwargs = super().get_form_kwargs(*args, **kwargs)
+        form_kwargs["instance"] = self.get_object().domain_info
+        return form_kwargs
+
+    def get_success_url(self):
+        """Redirect to the overview page for the domain."""
+        return reverse("domain-org-name-address", kwargs={"pk": self.object.pk})
```

Comment:
**question (non-blocking)**: I think this is probably right as it is, but I did wonder for a second if we wanted a successful save to land back on the domain overview page?